cumulus_library_catalog/catalog/icd10_annotated_counts.py

```diff
@@ -1,61 +1,93 @@
+            ("extension_code", None),
+            ("extension_str", None),
+        ]
+        if depth * 2 > len(potential_cols) | depth <= 0:
+            raise errors.CountsBuilderError(
+                f"Requested ICD10 depth expected to be between 1 and {len(potential_cols) / 2}"
+            )
+        annotation_cols = []
+        # if not at the tree root, get the parent tier for annotation
+        if depth != 1:
+            annotation_cols.append(potential_cols[(depth - 2) * 2])
+            annotation_cols.append(potential_cols[(depth - 2) * 2 + 1])
+        annotation_cols.append(potential_cols[(depth - 1) * 2 + 1])
+        annotation_cols.append(potential_cols[(depth - 1) * 2])
+        target_col = potential_cols[(depth - 1) * 2][0]
+
         if min_subject is None:
             min_subject = 10
         return self.count_patient(
-            table_name = "catalog__count_icd10_diagnoses",
-            table_cols=[
-                "code",
-            ],
+            table_name=table_name,
             source_table="core__condition",
+            table_cols=["code", "system"],
             where_clauses=[
                 # These codes are excluded from the catalog because they correspond
-                # to sensitive conditions (mental health/STDs) that hospitals 
+                # to sensitive conditions (mental health/STDs) that hospitals
                 # do not want to share with others for legal reasons
-                "block_code not in('A50-A64','A70-A74','B20-B20')",
-                "chapter_code not in('F01-F99','Z00-Z99')",
-                f"cnt_subject_ref >= {min_subject}"
-                ],
-            annotation= cumulus_library.CountAnnotation(
-                    field = 'code',
-                    join_table= '"umls"."icd10_hierarchy"',
-                    join_field = 'leaf_code',
-                    columns= ([    
-                        ('chapter_code', None),
-                        ('chapter_str', None),
-                        ('block_code', None),
-                        ('block_str', None),
-                        ('category_code', None),
-                        ('category_str', None),
-                        ('subcategory_1_code', None),
-                        ('subcategory_1_str', None),
-                        ('subcategory_2_code', None),
-                        ('subcategory_2_str', None),
-                        ('subcategory_3_code', None),
-                        ('subcategory_3_str', None),
-                        ('extension_code', None),
-                        ('extension_str', None),
-                    ])
-                ),
-            )
+                "block_code NOT IN ('A50-A64','A70-A74','B20-B20')",
+                "chapter_code NOT IN ('F01-F99','Z00-Z99')",
+                f"cnt_subject_ref >= {min_subject}",
+                f"{target_col} IS NOT NULL",
+                "system ='http://hl7.org/fhir/sid/icd-10-cm'",
```

Comment:
tiniest of nits: missing a space after the equal sign (here and when defining `catalog__count_icd10_all`)